### ISSUE TITLE
Optimize Hyperparameters and select the Best Model

### ISSUE THREAD

**Issue by ucaiado** [ucaiado/mlops-ml-pipeline-rental-prices#8]
- [ ] Run training with different hyperparameters using the Hydra system. On W&B there should be the results of several (>2) training jobs with different hyperparameters
- [ ] Select the best performing model and tag it as “prod”. Add the tag “prod” to the trained model with the best MAE.
- [ ] Verify test set performance is comparable to performance on the validation set (no overfitting). Implement the test_regression_model function in the main.py file. The test_regression_model is provided just as in the “data splitting” step.
